pyvista/core/cell.py

```diff
@@ -637,6 +638,20 @@ def _set_cells(self, cells: NumpyIntArray, n_cells: Optional[int], deep: bool) -
                 n_cells = cells.shape[0]
 
         self.SetCells(n_cells, vtk_idarr)
+
+        # Check that the size of cells array matches expectation.  Errors here
+        # are likely due to an invalid connectivity array. VTK SetCells can
+        # read memory outside of the array bounds if the padding is incorrect.
+        # See https://github.com/pyvista/pyvista/issues/5217
```

Comment:
I would encourage the code reader to read the PR description of the change rather than reading the code description. However, this is a personal policy and can be ignored.
```suggestion
        # https://github.com/pyvista/pyvista/pull/5404
```